### PR TITLE
3 get api

### PR DESCRIPTION
**Pull request by GowerRob**
Hello, please review my task 3 Get /api, which includes promisified readFile, thanks

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            expect(body.topics).toHaveLength(3);
+            body.topics.forEach((topic)=>{
+                expect(typeof topic.slug).toBe('string');
+                expect(typeof topic.description).toBe('string');
+            })
+        })
+    })
+})
+
+describe.only('/api',()=>{
```
Comment (pippaamy) [inline __tests__/app.test.js]: we need to take .only off before we push

File: __tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    test('get request will return a status code 200 and an object containing the api data',()=>{
+        return request(app)
+        .get('/api')
+        .expect(200)
+        .then((response)=>{
+            expect(typeof response.body).toBe('object')
+            expect(Object.keys(response.body).includes('GET /api/articles')).toBe(true)
```
Comment (pippaamy) [inline __tests__/app.test.js]: we need to make this more dynamic so you don't have to do a new test for each part that is added

File: app.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+const express = require('express');
+const {getTopics, getApi} = require("./db/controllers/controllers");
+const app = express();
+
+app.get('/api/topics', getTopics);
+app.get('/api', getApi);
+
+
+
+// app.use((err,req,res,next)=>{
```
Comment (pippaamy) [inline app.js]: lets delete this if its not used

File: db/controllers/controllers.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+exports.getTopics = (req,res)=>{
+    fetchTopics().then((topics)=>{
+        res.status(200).send({topics:topics});
+    }).catch((err)=>{
+        next(err)
+    });
+}
+
+exports.getApi = (req,res)=>{
+        readFile(`endpoints.json`,"utf-8")
```
Comment (pippaamy) [inline db/controllers/controllers.js:12]: nice!

File: __tests__/app.test.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+afterAll(()=>db.end());
+describe('/api/topics',()=>{
+    
+    test('GET 200 status coded return on get request',()=>{
+        return request(app)
+        .get('/api/topics')
+        .expect(200)
+    })
+
+    test('GET request should return an array of topic objects',()=>{
```
Comment (pippaamy) [inline __tests__/app.test.js]: im confused as to how I can see all your topic changes on your /api pr ??

**Review by pippaamy — COMMENTED** [review]
lets make those changes and do a new pr :) - remember no .only when pushing

File: __tests__/index.test.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        .get('/api')
+        .expect(200)
+        .then((response)=>{
+            expect(typeof response.body).toBe('object')
+            for(const endpoint in response.body){
+                expect(Object.keys(response.body[endpoint]).includes('description'))
+                expect(Object.keys(response.body[endpoint]).includes('queries'))
+                expect(Object.keys(response.body[endpoint]).includes('exampleResponse'))
+            }
+        })
```
Comment (NRMSMITH) [inline __tests__/index.test.js:42]: I would suggest having a look at the matcher .toMatchObject next time you need to look at dynamically checking objects :)

**Review by NRMSMITH — APPROVED** [review]
Great 🎉  Merge away 🚀

**Pull request merged**